examples/main.cpp

```diff
@@ -0,0 +1,222 @@
+   }
+   else if (mesh.GetNodes())
+   {
+      fec = mesh.GetNodes()->OwnFEC();
+      delete_fec = false;
+      cout << "Using isoparametric FEs: " << fec->Name() << endl;
+   }
+   else
+   {
+      fec = new H1_FECollection(order = 1, dim);
+      delete_fec = true;
+   }
+   FiniteElementSpace fespace(&mesh, fec);
+   cout << "Number of finite element unknowns: "
+        << fespace.GetTrueVSize() << endl;
+
+
+   // 5. Define the solution vector x as a finite element grid function
+   //    corresponding to fespace. Initialize x with initial guess of zero,
+   //    which satisfies the boundary conditions.
+   GridFunction x(&fespace);
+   x = 0.0;
+
+   Vector rhs(x.Size());
+   rhs.Randomize();
+
+   Vector x_ref = rhs;
+   Vector x_magma = x_ref;
+
+   // 6. Set up the bilinear form a(.,.) on the finite element space
+   const int NE = mesh.GetNE();
+   const int ndofs = x_ref.Size() / NE;
+
+   MassIntegrator mass_int;
+   Vector VecMassMats(ndofs * ndofs * NE);
+   mass_int.AssembleEA(fespace, VecMassMats, false);
+
+   //Batch solver...
+   DenseTensor LUMassMats(ndofs, ndofs, NE);
+   DenseTensor MassMats(ndofs, ndofs, NE);
+   Array<int> P(rhs.Size());
+
+   for (int i=0; i<ndofs * ndofs * NE; ++i)
+   {
+      LUMassMats.HostWrite()[i] = VecMassMats.HostRead()[i];
+      MassMats.HostWrite()[i] = VecMassMats.HostRead()[i];
+   }
+
+   //
+   //Compute reference solution
+   //
+   mfem::BatchLUFactor(LUMassMats, P);
+   mfem::BatchLUSolve(LUMassMats, P, x_ref);
+
+
+   //
+   //Compute Magma solution
+   //
+
+   //Compute magma version with variable batch solver
+   magma_int_t magma_device = 0;
+   magma_queue_t magma_queue;
+
+   magma_setdevice(magma_device);
+   magma_queue_create(magma_device, &magma_queue);
+
+
+   //Number of rows and columns of each matrix
+   Array<magma_int_t> num_of_rows(NE);
+   Array<magma_int_t> num_of_cols(NE);
+
+   //Pointers to mass matrices
+   Array<double *> magma_LUMassMats(NE);
+   for (int i=0; i<NE; ++i)
+   {
+      num_of_rows[i] = ndofs;
+      num_of_cols[i] = ndofs;
+      magma_LUMassMats.HostWrite()[i] = &MassMats.ReadWrite()[i*ndofs*ndofs];
+   }
+
+   Array<magma_int_t> ldda = num_of_cols;
+   Array<magma_int_t *> dipiv_array(NE);
+   Array<magma_int_t> info_array(NE);
+
+   //Pointers to pivot vectors
+   P.HostReadWrite();
+   P = 0.0;
+   for (int i=0; i<NE; ++i)
+   {
+      dipiv_array.HostWrite()[i] = &P.ReadWrite()[i*ndofs];
+   }
+
+   //Perform variable batch size factorization
+   magma_dgetrf_vbatched(num_of_rows.ReadWrite(), num_of_cols.ReadWrite(),
+                         magma_LUMassMats.ReadWrite(), ldda.ReadWrite(),
+                         dipiv_array.ReadWrite(), info_array.ReadWrite(),
+                         NE, magma_queue);
+
+
+   //
+   //Question: How do we perform the batch LU/Solve ?
+   //
+
+
+   //rhs only has 1 column
+   Array<int> rhs_num_of_cols(NE);
+   rhs_num_of_cols = 1;
+
+   //Question: Do we need to perform the pivoting ourselfs?
```

Comment:
Yes, I believe so.  The `dgetrs_batched` routine uses `magma_zlaswp_rowserial_batched` with the `dipiv_array`.  But, it looks like the `vbatched` version is slightly different ("left" and "right" versions).